iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/extensions/ReferenceTerminalsImpl.java

```diff
@@ -39,12 +39,14 @@ public ReferenceTerminalsImpl(Network network, Set<Terminal> terminals) {
                 Collections.nCopies(getVariantManagerHolder().getVariantManager().getVariantArraySize(), new LinkedHashSet<>()));
         setReferenceTerminals(terminals);
         this.referenceTerminalsListener = new ReferenceTerminalsListener();
-        network.addListener(this.referenceTerminalsListener);
+        // listeners are not (yet) supported on subnetworks
+        network.getNetwork().addListener(this.referenceTerminalsListener);
```

Comment:
I think that there is a problem if the extension is in a network **before** it is merged in another one.
In `AbstractNetwork.transferExtensions(…)`, the extension will be transferred from the `NetworkImpl` object (the network to merge) to the `SubnetworkImpl` (the merged subnetwork) via `Extension.setExtendable(…)`.
The listener should also be transferred from the old extendable to the new one in this method.

Reply:
Yes indeed there was a problem I was aware of, and I thought if was just impossible to solve, before you showed me the way, thanks again!

Reply:
Sorry, I hadn't understood that you encountered a problem here.
I'm glad the way we discussed together worked.
Thanks for the changes.